### ISSUE TITLE
Difficulty finding reason for unexpectedly bad results

### ISSUE THREAD

**Issue by Velkymoss** [HLR/DomiKnowS#421]
Hello guys,
I am facing a performance problem that I don't quite understand from domiknows side.
From studying the logs I could not find any hint during debugging unfortunately.
I did a baseline run with a pure pytorch set-up which results in 54% macro f1 on TB Dense for temporal event relation
extraction. With the same data, model and no constraints I get ~10% macro f1 on the test set with the Domiknows 
set-up.
From your experience, what is the best way to find out the reason for the bad performance from Domiknows side?

Here would be the key files of my pipeline:
graph: https://github.com/Velkymoss/TempQChainDomiknows/blob/main/src/tempQchain/graphs/graph_fr.py
program: https://github.com/Velkymoss/TempQChainDomiknows/blob/main/src/tempQchain/programs/program_fr.py
model: https://github.com/Velkymoss/TempQChainDomiknows/blob/main/src/tempQchain/programs/models.py
train+eval loop: https://github.com/Velkymoss/TempQChainDomiknows/blob/main/src/tempQchain/temporal_FR.py

I looked at the model output, which is producing valid logits. For the validation and test set,
the model seems to be stuck on predicting one label category only. Same behavior for constraints enabled or disabled.
Could it be that the sensor in the program are wrongly configured?

If it is not too much to ask, I would appreciate any hint of what I could further check to get to know the reasons for the 
unexpectedly bad preformance.

Thanks in advance for your effort!
